connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -3017,8 +3028,37 @@ class Dataset[T] private[sql] (
         .getStorageLevel)
   }
 
+  /**
+   * Defines an event time watermark for this [[Dataset]]. A watermark tracks a point in time
+   * before which we assume no more late data is going to arrive.
+   *
+   * Spark will use this watermark for several purposes: <ul> <li>To know when a given time window
```

Comment:
The html tags here look odd because of scalfmt changes. Otherwise, the comment is same as it for its scala-sql counterpart. 
cc: @zhenlineo

Reply:
I see. It is okay for now. If the scala doc looks different, we should address the formatting problem for the whole client codebase. It is out of this PR's scope to fix the formatting.